### PR TITLE
Various improvements and fixes

### LINKED ISSUE #207: Depviz web: alternative graph layouts
* [x] gantt
* [x] flow graph
* [ ] new engines

┆Issue is synchronized with this [Trello card](https://trello.com/c/g8sYzxFc)

### LINKED ISSUE #209: Add edge arrows
Dependency -> dependent

┆Issue is synchronized with this [Trello card](https://trello.com/c/FPGFELUx)

### LINKED ISSUE #214: Add debug box with metrics, last update, api quota, etc
┆Issue is synchronized with this [Trello card](https://trello.com/c/9eMhyhyM)

### LINKED ISSUE #217: UI/UX
Interesting stuff:

* Dat.gui (https://github.com/dataarts/dat.gui, http://workshop.chromeexperiments.com/examples/gui/)
* Tabler (https://github.com/tabler/tabler, https://tabler.io)
* Bootstrap Studio (https://bootstrapstudio.io/)

┆Issue is synchronized with this [Trello card](https://trello.com/c/Bfs6iWsi)

### LINKED ISSUE #261: Repair authentication
When switching to react, we lost the authentication feature

* header to send: https://github.com/moul/depviz/blob/v3.3.0/web/depviz.js#L42-L44
* know if auth is required: https://github.com/moul/depviz/blob/v3.3.0/web/depviz.js#L263-L269

I just configured another domain with auth enabled here: https://depviz-demo-auth.moul.io/ password=`d3pviz`

┆Issue is synchronized with this [Trello card](https://trello.com/c/VHqFebSo)

### PR DESCRIPTION
**Pull request by appigram**
This is PR contains refactored version of webapp (based on React Hooks) and first version of Gantt and Flow diagrams, a few issues exists:
- for example, repository `moul-bot/depviz-test`, API returns duplicate issues http://joxi.ru/n2YYn4ESbB5Pn2. so sometimes the diagrams are broken
- for the Gantt diagram duration required, I did 7 days by default, is it ok or it should be configured by a user, from issue description or smth else?
- another issue - missing nodes, for example when the Closed label is not checked. Diagrams will be broken for now

fixes #207 
fixes #217 
fixes #261 
fixes #214
fixes #209  

┆Issue is synchronized with this [Trello card](https://trello.com/c/j2DzSxB1)

### CODE REVIEW

**Comment by moul** [comment]
> for example, repository moul-bot/depviz-test, API returns duplicate issues http://joxi.ru/n2YYn4ESbB5Pn2. so sometimes the diagrams are broken

I will manage this from server side

> for the Gantt diagram duration required, I did 7 days by default, is it ok or it should be configured by a user, from issue description or smth else?

I will parse it, but it will never by mandatory, so we need to have a sane default, let's start with "7d"

> another issue - missing nodes, for example when the Closed label is not checked. Diagrams will be broken for now

This cannot be fixed, you should manage it from the web (see #184)

**Comment by moul** [comment]
I just fixed the "duplicate issues" problem here: https://github.com/moul/depviz/pull/260

will be in production in ~15 minutes

File: web/.env.dev [hunk 1]
```diff
@@ -2,3 +2,4 @@ NODE_ENV=development
 __DEV__=true
 #API_URL=https://depviz-demo.moul.io/api
 API_URL=https://depviz-demo-auth.moul.io/api/
+AUTH_TOKEN=d3pviz
```
Comment (moul) [inline web/.env.dev]: auth token should not be passed as env var but provided by the user with a `window.prompt`

Comment (appigram) [reply web/.env.dev]: @moul in latest commit auth token storing changed. Auth token saved to localstorage and we ask user to enter token in modal form

Comment (moul) [reply web/.env.dev]: :+1:

**Comment by moul** [comment]
* about codefactor issues, if I understand well, it's because of some vendor files? if so, can you setup a bower or equivalent package manager and gitignore the vendor files?
* FPS box is not available in production
* Missing the "redraw" input
* When switching to `gantt` layout, the website is broken with a javascript error
* the checkboxes does not work because the generated API URL use incorrect keys, i.e.:
    * https://depviz-demo.moul.io/api/graph?targets=moul/depviz&withClosed=false&withIsolated=false&withPrs=false&withoutExternal-deps=false&layout=grid, should be:
    * https://depviz-demo.moul.io/api/graph?targets=moul/depviz&withClosed=true&withoutIsolated=true&withoutPrs=true&withoutExternalDeps=true&layout=grid
	* accepted parameters: `targets`, `withClosed`, `withoutPrs`, `withoutExternalDeps`, `withoutIsolated`

**Comment by moul** [comment]
Hum finally sorry but I need that you make something **before** merging this

I saw that this PR introduced about 500 new files, can you use bower or equivalent to only keep our own code in this repo?

File: web/.env.dev [hunk 2]
```diff
@@ -1,3 +1,4 @@
 NODE_ENV=development
 __DEV__=true
-API_URL=https://depviz-demo.moul.io/api
+#API_URL=https://depviz-demo.moul.io/api
+API_URL=https://depviz-demo-auth.moul.io/api/
```
Comment (moul) [inline web/.env.dev]: This should not be hardcoded in the repo, feel free to create a new dedicated env file for testing authentication

but the default behavior should be to use unauthenticated depviz-demo

File: web/README.md [hunk 3]
```diff
@@ -1,46 +1,163 @@
+<h1 align="center">
```
Comment (moul) [inline web/README.md]: You should not copy the root README in the web folder

Use a README.md file that is dedicated to web or remove it completely

File: web/src/App.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+const App = () => {
+  const [showAuthModal, setShowAuthModal] = useState(!store.getItem('auth_token'))
+  const [authToken, setAuthToken] = useState(store.getItem('auth_token') || '')
+  const searchParams = new URLSearchParams(window.location.search)
+  const urlData = {
+    targets: searchParams.getAll('targets').join(',') || '',
+    withClosed: searchParams.get('withClosed') || '',
+    withoutIsolated: searchParams.get('withoutIsolated') || '',
+    withoutPrs: searchParams.get('withoutPrs') || '',
+    withoutExternalDeps: searchParams.get('withoutExternalDeps') || '',
```
Comment (moul) [inline web/src/App.js]: be careful with the variable prefixes

"withXXX" should be true if the checkbox is checked
"withoutXXX" shoule be false if the checkbox is checked

File: web/src/assets/img/flags/ad.svg [hunk 5]
```diff
@@ -0,0 +1 @@
....4l-1.8-.5h-.4c-.5 1.8-.7 2.7-1.6 4.2-.8 1.3-1.7 2.3-2.6 3l-.7.5zm5 18.2l.8-1.3 1.4-1.1h.4a8.7 8.7 0 0 1-.5 2.8l-.4 1-.5.5c-.5-.8-1.3-1.3-1.3-2zm33 1.8l1.4.6 1.5.9v.5l-1.5.2a8.4 8.4 0 0 1-1.3 0h-1l-.6-.4c.5-.7.8-1.6 1.4-1.8zm-9.8-2l1.4.5 1.5 1c0 .1.1.3 0 .4a9 9 0 0 1-2.7.3l-1-.1-.7-.3c.6-.7.9-1.7 1.5-1.8zm-17.4 2.1l1.5.5 1.5 1v.5a9 9 0 0 1-2.8.2h-1l-.6-.4c.5-.7.8-1.6 1.4-1.8zm-9-29.8c-.6-.3-1-1-.6-1.6.1-.2.4-.2.6-.4.2-.3.1-.5 0-.8l-.1-1-.2-1c0-.6 0-1 .4-1.6.2-.3.7-.6.8-.6.2.1 0 .5 0 .8 0 .5.1.7.3 1.2l.7 1.3c.2.6.4.8.4 1.4 0 .5 0 .7-.2 1.2a2 2 0 0 1-.6.8 2 2 0 0 1-.8.4 1.1 1.1 0 0 1-.6 0z"/></g><use width="100%" height="100%" y="36.6" xlink:href="#b"/></g><path fill="none" stroke="#703d29" stroke-width=".5" d="M264.1 175.5h52.6V228h-52.6zm58.2 0h52.6V228h-52.6zm-58 98c.1 1 .5 2.6 1.4 4.3 1 1.5.6 1.4 2.7 3.8a15.3 15.3 0 0 0 4 2.9 32.7 32.7 0 0 0 15 2.6c2.7-.1 4.8-.4 6.6-.7a71 71 0 0 1 11-.6c1.5 0 3 .3 4.7.6 3.5.7 7 2 7 2v-54.7h-52.6V271l.2 2.4zm110.4 0a13 13 0 0 1-1.4 4.3c-1 1.5-.6 1.4-2.7 3.8a15.4 15.4 0 0 1-4 2.9c-1.3.7-2.3 1-4.4 1.6a32.6 32.6 0 0 1-10.6 1c-2.7-.1-4.8-.5-6.5-.7a71 71 0 0 0-7.2-.6 40.5 40.5 0 0 0-3.9 0c-1.5 0-3 .3-4.7.6-3.5.7-7 2-7 2v-54.8H375v37.5l-.2 2.4z"/></svg>
```
Comment (moul) [inline web/src/assets/img/flags/ad.svg]: use bower as much as you can to manage vendored assets

**Pull request merged**